### DISCUSSION TITLE
Why cannot add bloom filter on the common columns of agg key table?

### DISCUSSION THREAD

**Discussion by m3ngzhang** [apache/doris#13958] [A - General / Q&A] [answered]
Like my title, I tried to add bloom filter to one of my agg table column, in order to obtain faster query performance. But I eventually got the error message like "The bloom filter can only be used to unique key/dup key table or the key column of agg key table ". I checked the paper work of Doris and find out that the bloom filter is created on the data page  level. I do not understand the difference between three table models in the data page structure. I think they should be the same, which makes me really confused when I encountered such error.

**Comment by m3ngzhang** [comment]
Here is my error msg.
Bloom filter index only used in columns of UNIQUE_KEYS/DUP_KEYS table or key columns of AGG_KEYS table. invalid column: BASE_POP_MOPHONE

**Comment by zy-kkk** [comment, marked as answer]
In the case of bloom_filter, in the duplicate and unique models, you can create it in any column, but in the aggregate model, you can only create it in the key column, not the value column.

**Reply by m3ngzhang** [reply]
I got that idea. Can you explain why?

**Reply by zy-kkk** [reply]
The value column of the aggregate table cannot be used as a conditional column

**Reply by m3ngzhang** [reply]
Thanks for replay. I mean if there is bloom filter on the value column, it can filter out a large number of Pages at query time. And this is what happens in the duplicate and unique models. Right? Why doesn't doris  do this to agg table? If there is something wrong with my understanding, please point it out.

**Reply by zy-kkk** [reply]
For the aggregate model, you can think of filter as a `group+having` operation on the value column, but bloom_filter only applies to normal `where =` or `where in` operations
